src/Validation/URLValidationRESTController.php

```diff
@@ -0,0 +1,303 @@
+<?php
+/**
+ * REST endpoint providing theme scan results.
+ *
+ * @package AMP
+ * @since 2.1
+ */
+
+namespace AmpProject\AmpWP\Validation;
+
+use AMP_Validated_URL_Post_Type;
+use AMP_Validation_Error_Taxonomy;
+use AmpProject\AmpWP\DevTools\UserAccess;
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Error;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * URLValidationRESTController class.
+ *
+ * @since 2.1
+ */
+final class URLValidationRESTController extends WP_REST_Controller implements Delayed, Service, Registerable {
+
+	/**
+	 * The URL validation context in the editor.
+	 *
+	 * @var string
+	 */
+	const CONTEXT_EDITOR = 'amp-editor';
```

Comment:
I think that this `context` may not be necessary. If we want to limit the fields, we can rather use the [`_fields` param](https://developer.wordpress.org/rest-api/using-the-rest-api/global-parameters/#_fields).  As of WordPress 5.3, this included the ability to [filter via nested fields](https://make.wordpress.org/core/2019/10/10/filtering-nested-rest-response-_fields-in-wp-5-3/).

Reply:
Removed in 2bff80018.